pkg/controller/test/actions.go

```diff
@@ -79,19 +79,14 @@ func (a *action) Action() coretesting.Action {
 
 // Matches compares action.action with another Action.
 func (a *action) Matches(act coretesting.Action) error {
-	matches := reflect.DeepEqual(a.action, act)
-	if matches {
-		return nil
+	diff := cmp.Diff(a.action, act,
+		cmp.FilterPath(func(p cmp.Path) bool {
```

Comment:
```suggestion
		// We ignore differences in .ManagedFields since the expected object does not have them.
		// FIXME: don't ignore this field
		cmp.FilterPath(func(p cmp.Path) bool {
```